src/erlfmt.erl

```diff
@@ -220,3 +255,51 @@ format_error({not_equivalent, Node1, Node2}) ->
     );
 format_error(could_not_reparse) ->
     "formatter result invalid, could not reparse".
+
+verify_ranges(Forms, StartLocation, EndLocation) ->
+    case possible_ranges(Forms, StartLocation, EndLocation) of
+        [{StartLocation, EndLocation}] ->
+            ok;
+        Options ->
+            {options, Options}
+    end.
+
+possible_ranges(Forms, StartLocation, EndLocation) ->
+    case forms_in_range(Forms, StartLocation, EndLocation) of
+        [] ->
+            [];
+        [OnlyForm] ->
+            [get_location_range(OnlyForm)];
+        MultipleForms ->
+            combine(
+                get_possible_locations(MultipleForms, StartLocation, fun get_location/1),
+                get_possible_locations(lists:reverse(MultipleForms), EndLocation, fun get_end_location/1))
```

Comment:
I'm not sure I understand correctly what is happening here. For successful operation we only accept ranges that overlap exactly one form, but when there are multiple overlapping forms we return as options all the possible ranges overlapping all of them. Is that correct? If we only accept ranges with one form, here we should return only ranges that contain one form.

Reply:
So we allow ranges that exactly contains 1 or more form, so the options are the possible forms you could have meant to include say for example that you have a range with three forms, where the range starts in form 1, form 2 is fully contained in the range and form 3 ends after the range so you have two sensible start locations the start of form 1 and the start of form 2, and two sensible end locations the end of form 2 and the end of form 3.
I.e. you end up with four possible ranges {1,2,3}, {2,3}, {1,2} and {2}.